src/AutorestSwift/Models/Complex/SchemaResponse.swift

```diff
@@ -43,15 +43,22 @@ class SchemaResponse: Response {
     public required init(from decoder: Decoder) throws {
         let container = try decoder.container(keyedBy: CodingKeys.self)
 
-        if let dictionarySchema = try? container.decode(DictionarySchema.self, forKey: .schema) {
-            self.schema = dictionarySchema
-        } else if let arraySchema = try? container.decode(ArraySchema.self, forKey: .schema) {
-            self.schema = arraySchema
-        } else if let objectSchema = try? container.decode(ObjectSchema.self, forKey: .schema) {
-            self.schema = objectSchema
-        } else {
-            self.schema = try container.decode(Schema.self, forKey: .schema)
+        let schemaContainer = try container.nestedContainer(keyedBy: Schema.CodingKeys.self, forKey: .schema)
+        let type = try? schemaContainer.decode(AllSchemaTypes.self, forKey: Schema.CodingKeys.type)
+
+        var schema: Schema?
+        switch type {
+        case .array:
+            schema = try? container.decode(ArraySchema.self, forKey: .schema)
+        case .dictionary:
+            schema = try? container.decode(DictionarySchema.self, forKey: .schema)
+        case .object:
+            schema = try? container.decode(ObjectSchema.self, forKey: .schema)
+        default:
+            schema = try container.decode(Schema.self, forKey: .schema)
         }
```

Comment:
Could we wrap this logic into a Schema initializer? That would make it more reusable. 
`init(withType type: AllSchemaTypes, andContainer container: KeyedContainer)`

Reply:
Create a common generic function to decode Schema. It needs to be generic because we need to pass the CodingKeys of different container class which has `schema` as its property.